### ISSUE TITLE
posix_spawn(3) support on OpenBSD

### ISSUE THREAD

**Issue by nuudlman** [rust-lang/libc#3501]
OpenBSD has supported posix_spawn(3) since [version 5.2](https://github.com/openbsd/src/blob/49a7a16f751fdb69d77674b1b08e5645c78f12da/lib/libc/gen/posix_spawn.c)

It appears that support is already plumbed up for NetBSD, so I'll try to set up a pull request later today.

**Comment by nuudlman** [comment]
OpenBSD defines `posix_spawnattr_t` and `posix_spawn_file_actions_t` opaquely, so I'm not sure how to define those symbols. 

Right now I have it set up as 
```rust
pub type posix_spawnattr_t = *mut __posix_spawnattr;
pub type posix_spawn_file_actions_t = *mut __posix_spawn_file_actions;
``` 
but I can't get a definition for `__posix_spawnattr` and `__posix_spawn_file_actions` that will pass the either the compiler or the test suite.

If I define the types exactly as they actually are, the test suite fails because it tries to instantiate them directly. Defining an empty type (`pub struct __posix_spawnattr {}`) runs into the missing copy implementations lint.

Is there something obvious I'm missing here?

**Comment by semarie** [comment]
opaque type could be implemented using:

```rust
pub type posix_spawnattr_t = *mut ::c_void;
pub type posix_spawn_file_actions_t = *mut ::c_void;
```

it means there are "pointer on something".

this strategy is already used for `pthread_attr_t` for example (which is also opaque type on OpenBSD).

**Comment by nuudlman** [comment]
That worked! Thanks for the suggestion.

**Comment by nuudlman** [comment]
Implemented in #3502
